File: face/geomajas-face-dojo/client/src/main/resources/geomajas/map/store/LayerStore.js
dojo.provide("geomajas.map.store.LayerStore");
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2011 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */

dojo.declare("LayerStore", null, {

	/**
	 * @class Definition of a FeatureStore. Feature collection handling...
	 * @author Pieter De Graef
	 *
	 * @constructor
	 */
	constructor : function () {
	},

	/**
	 * Does this store currently contain the feature with given id?
	 * @param id A feature's identifier.
	 * @returns Return true or false.
	 */
	contains : function (/*String*/id) {
		return false;
	},

	/**
	 * Add a new feature to the underlying datastructure.
	 * @param element Feature object.
	 */
	addElement : function (element) {
	},

	/**
	 * Remove a feature from the underlying datastructure. (if it's in there)
	 * @param element Feature object.
	 */
	removeElement : function (element) {
	},

	clear : function () {
	}

});

File: client/geomajas-client-smartgwt/client/src/main/java/org/geomajas/smartgwt/client/map/store/LayerStore.java
/*
 * This is part of Geomajas, a GIS framework, http://www.geomajas.org/.
 *
 * Copyright 2008-2013 Geosparc nv, http://www.geosparc.com/, Belgium.
 *
 * The program is available in open source according to the GNU Affero
 * General Public License. All contributions in this program are covered
 * by the Geomajas Contributors License Agreement. For full licensing
 * details, see LICENSE.txt in the project root.
 */

package org.geomajas.smartgwt.client.map.store;

import java.util.Collection;

import org.geomajas.smartgwt.client.map.cache.tile.Tile;

/**
 * A store is a collection of tiles for a particular layer.
 *
 * @param <T>
 *
 * @author Pieter De Graef
 */
public interface LayerStore<T extends Tile> {

	/** Empty the entire <code>LayerStore</code> unconditionally. */
	void clear();

	/**
	 * Is the <code>LayerStore</code> out of sync ?
	 *
	 * @return true if dirty
	 */
	boolean isDirty();

	/** Return an array of all the <code>SpatialNode</code>s currently present in the cache. */
	Collection<T> getTiles();
}
